<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1129</width>
    <height>648</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QUdpHexTerminal</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Адрес: </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="eAddress"/>
        </item>
        <item>
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>наш порт: </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="eOurPort">
          <property name="minimumSize">
           <size>
            <width>80</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximum">
           <number>65535</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>порт цели: </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="eTargetPort">
          <property name="minimumSize">
           <size>
            <width>80</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximum">
           <number>65535</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="bConnect">
          <property name="text">
           <string>Bind</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lOK">
          <property name="text">
           <string>No Connection</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lNBytes">
          <property name="text">
           <string>read 0x0000 bytes</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>sent 0x0000 bytes</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="Line" name="line">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTabWidget" name="tabWidget">
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="tabHex">
           <attribute name="title">
            <string>Hex</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <property name="spacing">
             <number>2</number>
            </property>
            <property name="leftMargin">
             <number>2</number>
            </property>
            <property name="topMargin">
             <number>2</number>
            </property>
            <property name="rightMargin">
             <number>2</number>
            </property>
            <property name="bottomMargin">
             <number>2</number>
            </property>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <item>
                 <widget class="QPushButton" name="bClearTransmitter">
                  <property name="text">
                   <string>Clear</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="bSend">
                  <property name="text">
                   <string>Send</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="bReSend">
                  <property name="text">
                   <string>Send &amp;&amp; Repeat</string>
                  </property>
                  <property name="checkable">
                   <bool>true</bool>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_2">
                  <property name="text">
                   <string>every</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="eReSendTimeMs">
                  <property name="minimumSize">
                   <size>
                    <width>50</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="maximum">
                   <number>60000</number>
                  </property>
                  <property name="value">
                   <number>300</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label">
                  <property name="text">
                   <string>ms</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QHexEdit" name="eSender" native="true"/>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tabScripts">
           <attribute name="title">
            <string>Scripts</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="0,0,0,0">
              <item>
               <widget class="QPushButton" name="bAddScript">
                <property name="maximumSize">
                 <size>
                  <width>45</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>Add</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="bSaveScripts">
                <property name="maximumSize">
                 <size>
                  <width>50</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>Save</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="bClearScripts">
                <property name="maximumSize">
                 <size>
                  <width>45</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="text">
                 <string>Clear</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,1">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QPushButton" name="bClearReceiver">
              <property name="text">
               <string>Clear</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QHexEdit" name="eReceiver" native="true"/>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QHexEdit</class>
   <extends>QWidget</extends>
   <header>QHexEdit/qhexedit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
